fms/fms_io.F90

```diff
@@ -2506,7 +2507,15 @@ subroutine save_restart(fileObj, time_stamp, directory, append, time_level)
      if (PRESENT(time_stamp)) then
         if(len_trim(restartname)+len_trim(time_stamp) > 79) call mpp_error(FATAL, "fms_io(save_restart): " // &
           "Length of restart file name + time_stamp is greater than allowed character length of 79")
-        restartname = trim(time_stamp)//"."//trim(restartname)
+           has_dot = .false.
+           do i=1,len(trim(time_stamp))
+              if (time_stamp(i:i) == ".") has_dot = .true.
```

Comment:
I think this should be `if ( trim(time_stamp)(i:i)  ..etc` or just remove the trim in the do loop, since the lengths wouldn't match up if any whitespace was trimmed